src/status_im/navigation/events.cljs

```diff
@@ -171,3 +171,30 @@
   (let [view-id (if (= view-id :shell-stack) (shell.utils/calculate-view-id) view-id)]
     {:db             (assoc db :view-id view-id)
      :set-view-id-fx view-id}))
+
+(defn- wizard-find-next-screen
+  [db flow-config current-screen]
+  (->> flow-config
+       (filter (fn [{:keys [skip-step? screen-id]}]
+                 (and (not= screen-id current-screen)
+                      (not (and (fn? skip-step?) (skip-step? db))))))
+       first))
+
+(rf/reg-event-fx
+ :navigation/wizard-forward
+ (fn [{:keys [db]} [{:keys [current-screen flow-config start-flow?]}]]
+   (let [next-screen (wizard-find-next-screen db flow-config current-screen)]
+     {:fx [[:dispatch
+            (if start-flow?
+              [:open-modal (:screen-id next-screen)]
+              [:navigate-to-within-stack [(:screen-id next-screen) current-screen]])]]})))
+
+(rf/reg-event-fx
+ :navigation/wizard-backward
+ (fn [{:keys [db]}]
+   (let [stack          (:modal-view-ids db)
+         current-screen (last stack)]
+     {:fx [[:dispatch
+            (if (= (count stack) 1)
```

Comment:
Wonder if we can check if current screen is the first in the flow without using `modal-view-ids`. After merging #19163 I think we can use `:view-id` and compare it with the first screen in the flow that is passed in `:navigation/wizard-forward` when `start-flow?` is `true` (we may need a new key like `:wizard/initial-screen` to store it). If that works we could think of removing `:modal-view-ids` cc @smohamedjavid @Parveshdhull

Reply:
![image](https://github.com/status-im/status-mobile/assets/17097240/a9cd2fb6-0384-48e4-a969-caa6f80351cb)

These are exactly same.
We should get rid of `:navigate-back-within-stack` and just have `:navigate-back`. And navigate-back should handle this logic of popping the screen based on where we are in navigation. (within a stack or pushing in root)

Reply:
And if we directly want to pop to a specific screen, we can just use. :navigate-back-to

Reply:
So, in short we don't need `:navigation/wizard-backward`, this should be handled by navigation itself. (what to pop, pushed screen or root itself).

navigate-within-stack is new functionality we introduced for on-boarding. Its still a little immature, and we need to improve it. We should modify navigation code itself according to this feature.

Reply:
sounds good @Parveshdhull, ideally we don't block this pr for too long so let's all be sure to review Brian's pr (19163). Additionally if there are issues for 2 & 3 please reach out if any help is needed with implementing, reviewing or any other steps of the process 👍

Reply:
Thank you @J-Son89,
I think Brian's PR is ready to merge, as soon as e2e tests are complete.
I will create issues for task 2 & 3, will complete by Monday.

Reply:
issue: https://github.com/status-im/status-mobile/issues/19254

Reply:
@Parveshdhull I agree with the approach of getting rid of `:navigate-back-within-stack` and handle both cases in `:navigate-back`. Currently `:navigate-back` always dismisses the current modal, so we should refactor that logic if we want it to support navigating back within modals. Maybe we should remove [this line](https://github.com/status-im/status-mobile/blob/6bda54266ba496170a7d8862a9c2a71b524c7d81/src/status_im/navigation/effects.cljs#L100) and if we want to dismiss a modal, explicitly dispatch `:dismiss-modal` event, or keep that call only if `:view-id` it is the first screen in the stack. If the last option is not possible, maybe we would still need `:navigation/wizard-backward` which should decide wether to dispatch `:navigate-back` or `:dismiss-modal`. If we can handle that within `:navigate-back` in a generic way, we could definitely remove `:navigation/wizard-backward`.

Reply:
Completed 2&3, so this PR can be unblocked - https://github.com/status-im/status-mobile/pull/19274

Reply:
Done